### PR TITLE
Kerberos fixes

### PR DESCRIPTION
**Pull request by hardening**
This PR addresses bugs with Kerberos:
* Encrypt/Decrypt message were failing
* when kerberos was failing in Negociate, the FreeRDP was crashing

### CODE REVIEW

**Review by akallabeth — APPROVED** [review]
LGTM

File: winpr/libwinpr/sspi/Kerberos/kerberos.c [hunk 1]
```diff
@@ -643,7 +656,11 @@ static SECURITY_STATUS SEC_ENTRY kerberos_EncryptMessage(PCtxtHandle phContext,
 	sspi_gss_buffer_desc input;
 	sspi_gss_buffer_desc output;
 	PSecBuffer data_buffer = NULL;
+	PSecBuffer sig_buffer = NULL;
 	context = (KRB_CONTEXT*)sspi_SecureHandleGetLowerPointer(phContext);
+	BYTE *ptr;
```
Comment (akallabeth) [inline winpr/libwinpr/sspi/Kerberos/kerberos.c:661]: Better move that above `context = (KRB_CONTEXT*)sspi_SecureHandleGetLowerPointer(phContext);`, VS2010 is nasty...

**Comment by akallabeth** [comment]
@hardening missed that on first check, the formatting seems wrong. better run `clang-format`

File: winpr/libwinpr/sspi/Kerberos/kerberos.c (hunk 1 above)
Comment (hardening) [reply winpr/libwinpr/sspi/Kerberos/kerberos.c]: done

**Pull request merged**